pkg/server/module_server.go

```diff
@@ -105,18 +103,22 @@ func (s *ModuleServer) Run() error {
 		return err
 	}
 
-	// Start dskit modules.
-	s.childRoutines.Go(func() error {
-		err := s.moduleService.Run(s.context)
-		if err != nil && !errors.Is(err, context.Canceled) {
-			return err
-		}
-		return nil
-	})
-
 	s.notifySystemd("READY=1")
 	s.log.Debug("Waiting on services...")
-	return s.childRoutines.Wait()
+
+	m := modules.New(s.cfg.Target)
+
+	m.RegisterModule(modules.Core, func() (services.Service, error) {
```

Comment:
it seems like this isn't the best place to put module registration long term, but it works for now to avoid some circular dependency issues.

Reply:
This is either a note for myself, or for us to talk about together when you're back: I didn't think this module server itself would be a dskit module - (I thought) it is the engine that manages and launches the dskit modules. This does feel a bit overcomplicated 🤔